template/README.md

```diff
@@ -522,6 +538,191 @@ Then, on the server, regardless of the backend you use, you can read `index.html
 
 If you use a Node server, you can even share the route matching logic between the client and the server. However duplicating it also works fine in simple cases.
 
+## Running Tests
+
+>Note: this feature is available with `react-scripts@0.3.0` and higher.
+
+Create React App uses [Jest](https://facebook.github.io/jest/) as its test runner. To prepare for this integration, we did a [major revamp](https://facebook.github.io/jest/blog/2016/09/01/jest-15.html) of Jest so if you heard bad things about it, give it another try.
+
+Jest is a Node-based runner. This means that the tests always run in a Node environment and not in a real browser. This lets us enable fast iteration speed and prevent flakiness.
```

Comment:
Shall we add "If you'd like to run real end-to-end tests, try selenium" or something? We should totally add a photo of the testing pyramid.